### PR TITLE
Improve documentation of `Vec::split_off(...)`

### PR DESCRIPTION
**Pull request by mqudsi**
The previous ordering of the sentences kept switching between the return
value and the value of `self` after execution, making it hard to follow.

Additionally, as rendered in the browser, the period in "`Self`. `self`"
was difficult to make out as being a sentence separator and not one code
block.

### CODE REVIEW

File: src/liballoc/vec.rs [hunk 1]
```diff
@@ -1293,10 +1293,10 @@ impl<T> Vec<T> {
 
     /// Splits the collection into two at the given index.
     ///
-    /// Returns a newly allocated `Self`. `self` contains elements `[0, at)`,
-    /// and the returned `Self` contains elements `[at, len)`.
-    ///
-    /// Note that the capacity of `self` does not change.
+    /// Copies the range from `[at, len)` to a newly allocated `Self`
```
Comment (dtolnay) [inline src/liballoc/vec.rs]: Note that these are moves not copies, as the Copy trait is not involved.

**Review by dtolnay — COMMENTED** [review]
Hmm, I find the original wording clearer here, although neither is great and I agree that the ``` `Self`. `self` ``` part comes out weird.

I think something like this would work best:

> Returns a newly allocated vector containing elements `[at, len)` of `self`. The original vector will be left holding elements `[0, at)`, with its capacity unchanged.

**Comment by JohnCSimon** [comment]
Ping from triage.
@mqudsi , please address the comments from @dtolnay 
Thank you!

**Comment by mqudsi** [comment]
What would be the best way to make it clear that this is not a "cheap" operation (my intention behind the (incorrect) usage of "copies" in the text)?

Since the original vector retains its previous capacity (which necessarily overlaps in memory with the range being split off), an allocation at least equivalent to `[at, len)` must be made.

**Comment by dtolnay** [comment]
I think "newly allocated vector" in my recommended wording is sufficient to indicate that there is a cost. I don't know that the cost is worth emphasizing beyond that.

**Comment by JohnCSimon** [comment]
Ping from triage:
@dtolnay is this PR ready for review - can it be approved for merging?
cc: @mqudsi 
Thanks

**Comment by dtolnay** [comment]
It looks like nothing has happened since my last review.

**Comment by mqudsi** [comment]
I'm sorry, I received the notifications about force pushes (presumably from rebasing) and thought that someone had taken over the issue.

I have pushed an update as requested.

**Comment by dtolnay** [comment]
Sorry about the confusion -- @Dylan-DPC had pushed something to your branch, which I reverted. @Dylan-DPC please don't do that. It didn't even address the earlier review.

**Review by dtolnay — APPROVED** [review]
Thanks! I am happy with the new wording.

**Comment by mqudsi** [comment]
Thanks for reviewing and for helping with the wording!

fwiw, I personally didn't like all the `self` and `Self` in the original wording (rather than using the word "vector") but assumed that it was there for a reason or by convention. Glad it's not the case ☺
